jsonnet/kube-prometheus/components/kube-state-metrics.libsonnet

```
@@ -46,6 +46,7 @@ local defaults = {
       runbookURLPattern: 'https://runbooks.prometheus-operator.dev/runbooks/kube-state-metrics/%s',
     },
   },
+  upstreamProbesOptIn:: false,
```

a comment explaining the purpose + implementation would be useful here.
```suggestion
  enableProbes:: false,
```